src/components/data-processing/GenesVsUMIs/CalculationConfig.jsx

```diff
@@ -1,24 +1,63 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import {
   Space,
   InputNumber,
   Form,
   Tooltip,
   Select,
+  Slider,
+  Alert,
+  Radio,
+  Button,
 } from 'antd';
+import { useDispatch } from 'react-redux';
 import { InfoCircleOutlined } from '@ant-design/icons';
+import useUpdateThrottled from 'utils/customHooks/useUpdateThrottled';
+import { runQC } from 'redux/actions/pipeline';
 
 const { Option } = Select;
 
 const GenesVsUMIsConfig = (props) => {
+  const dispatch = useDispatch();
   const {
-    // eslint-disable-next-line no-unused-vars
-    config, disabled, plotType, updateSettings,
+    config, updateSettings, disabled, rerunRequired, experimentId,
   } = props;
+  const [newConfig, handleChange] = useUpdateThrottled(updateSettings, config);
+
+  // if prediction interval is not set - use the passed p-value from the pipeline
+  useEffect(() => {
+    if (!config.predictionInterval) {
+      updateSettings(
+        { predictionInterval: parseFloat(1 - config.regressionTypeSettings[config.regressionType]['p.level']).toFixed(6) },
+        false,
+      );
+    }
+  }, []);
```

Comment:
This only run once, when the component is loaded, and it checks if a property `predictionInterval` is set in `config`, which by default does not exist. There is no guard in the plot which prevents this component from rendering before the `config` is loaded. Wouldn't this codeblock always run and `updateSettings` always called?

Reply:
I think the update of this value should not be done here, but when generating plot spec.

Reply:
good idea, I'll think of adding it there, we might even not need that other change checking for outstandingChanges that I also added for this.